pkg/systemlogmonitor/log_monitor.go

```diff
@@ -165,7 +188,21 @@ func (l *logMonitor) generateStatus(logs []*logtypes.Log, rule systemlogtypes.Ru
 				break
 			}
 		}
+		if *l.config.EnableMetricsReporting {
+			err := problemmetrics.GlobalProblemMetricsManager.SetProblemGauge(rule.Condition, rule.Reason, true)
+			if err != nil {
+				glog.Errorf("Failed to update problem gauge metrics for problem %q, reason %q: %v",
+					rule.Condition, rule.Reason, err)
+			}
+		}
+	}
+	if *l.config.EnableMetricsReporting {
+		err := problemmetrics.GlobalProblemMetricsManager.IncrementProblemCounter(rule.Reason, 1)
```

Comment:
This needs to put back into line 167 and line 184. For condition, we only generate conditions when the if check on line 176 is true.

Reply:
Done.